### ISSUE TITLE
linter: `react/react-compiler` throws error on a `for` loop without a variable-declaration initializer

### ISSUE THREAD

**Issue by mds-ant** [oxc-project/oxc#24842]
### What version of Oxlint are you using?

1.75.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["react"],
  "rules": {
    "react/react-compiler": "error"
  }
}
```

## Repro

```tsx
import { useEffect } from "react";

function Component() {
  useEffect(() => {
    let i = 0;
    for (; i < 3; i++) {
      console.log(i);
    }
  });
  return null;
}
```

## Actual output

```
  x react(react-compiler): Invariant: Expected a variable declaration
    ,-[repro.tsx:3:1]
  2 |     
  3 | ,-> function Component() {
  4 | |     useEffect(() => {
  5 | |       let i = 0;
  6 | |       for (; i < 3; i++) {
  7 | |         console.log(i);
  8 | |       }
  9 | |     });
 10 | |     return null;
 11 | `-> }
    `----

Found 0 warnings and 1 error.
```
